client/src/components/CreatePost.tsx

```diff
@@ -0,0 +1,225 @@
+
+  const [openPopup, setOpenPopup] = useState(false);
+  const [openPopupError, setOpenPopupError] = useState(false);
+
+  const allFields = [title, body, file];
+
+  // handle functions
+  const handleClickOpen = () => {
+    let noEmptyFields = true;
+    allFields.forEach((inputField) => {
+      if (inputField === "") {
+        noEmptyFields = false;
+      }
+    });
+    if (noEmptyFields) {
+      setOpenPopup(true);
+    } else {
+      setOpenPopupError(true);
+    }
+    return;
+  };
+
+  const handleClose = () => {
+    setOpenPopup(false);
+  };
+
+  // handle functions
+  const handleImageUpload = (event: React.ChangeEvent<{}>) => {
+    const target = event.target as HTMLInputElement;
+    /** do something with the file **/
+    let url = URL.createObjectURL((target.files as FileList)[0]);
+    setFile(url);
+    console.log(url);
+  };
+
+  return (
+    <ThemeProvider theme={theme}>
+      <Container component="main" maxWidth="md">
+        <CssBaseline />
+        <Box
+          sx={{
+            marginTop: 8,
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "center",
+          }}
+        >
+          <Typography component="h3" variant="h3" fontWeight="bold">
+            Create Post
+          </Typography>
+          {/* form  begins*/}
+          <Box component="form" noValidate sx={{ mt: 2 }}>
+            <Grid container spacing={2}>
+              <Grid item xs={12}>
+                <TextField
+                  required
+                  fullWidth
+                  label="Title"
+                  size="small"
+                  onChange={(e) => setTitle(e.target.value)}
+                />
+              </Grid>
+
+              <Grid item xs={12}>
+                <TextField
+                  required
+                  fullWidth
+                  label="Body"
+                  multiline
+                  rows={6}
+                  onChange={(e) => setBody(e.target.value)}
+                />
+              </Grid>
+
+              <Grid item xs={12} md={4}>
+                <TextField
+                  required
+                  label="Upload Thumbnail"
+                  type="file"
+                  fullWidth
+                  InputLabelProps={{ shrink: true }}
+                  size="small"
+                  onChange={(e) => handleImageUpload(e)}
+                />
+              </Grid>
+
+              <Grid item xs={12} md={3}>
+                <TextField
+                  label="Event Capacity"
+                  placeholder="40"
+                  fullWidth
+                  size="small"
+                  onChange={(e) => setEventCapacity(e.target.value)}
+                />
+              </Grid>
+
+              <Grid item xs={12} md={5}>
+                <TextField
+                  fullWidth
+                  label="Location"
+                  placeholder="Zoom: https://utoronto.zoom.us/j/7227206933#success"
```

Comment:
I would advise against this type of a placeholder. Perhaps just "Deerfield Hall" or something?